source/admin_manual/list_of_events.rst

```diff
@@ -2378,561 +1519,525 @@ obox plugin
+   :field error: Error message if the refresh failed.
+
+
+.. dovecot_core:event:: metacache_user_refresh_started
+   :added: v2.3.11
+   :inherit: mail_user, metacache_refresh
+
+   Metacache is being refreshed when user is being accessed. Sent only when a
+   storage operation is done to perform the refresh; the event isn't sent if
+   the metacache is used without refreshing.
+
+
+.. dovecot_core:event:: metacache_user_refresh_finished
+   :added: v2.3.11
+   :inherit: mail_user, metacache_refresh
+
+   Metacache was refreshed when user is being accessed. Sent only when a
+   storage operation is done to perform the refresh; the event isn't sent if
+   the metacache is used without refreshing.
+
+
+.. dovecot_core:event:: metacache_mailbox_refresh_started
+   :added: v2.3.11
+   :inherit: mailbox, metacache_refresh
+
+   Metacache is being refreshed when mailbox is being accessed. Sent only when
+   a storage operation is done to perform the refresh; the event isn't sent if
+   the metacache is used without refreshing.
+
+
+.. dovecot_core:event:: metacache_mailbox_refresh_finished
+   :added: v2.3.11
+   :inherit: mailbox, metacache_refresh
+
+   Metacache was refreshed when mailbox is being accessed. Sent only when
+   a storage operation is done to perform the refresh; the event isn't sent if
+   the metacache is used without refreshing.
+
+
+.. dovecot_event:field_group:: metacache_bundle_download
+
+   :field filename: Bundle filename.
+   :field bundle_type: Bundle type: ``diff``, ``base``, or ``self``.
+   :field bundle_size: Size of the bundle file in bytes (uncompressed).
+   :field error: Error message if the download failed.
+
+
+.. dovecot_core:event:: metacache_user_bundle_download_started
+   :added: v2.3.11
+   :inherit: mail_user, metacache_bundle_download
+
+   User index bundle file is being downloaded (can happen while the user is
+   being refreshed).
+
+
+.. dovecot_core:event:: metacache_user_bundle_download_finished
+   :added: v2.3.11
+   :inherit: mail_user, metacache_bundle_download
+
+   User index bundle file was downloaded (can happen while the user is being
+   refreshed).
+
+
+.. dovecot_core:event:: metacache_mailbox_bundle_download_started
+   :added: v2.3.11
+   :inherit: mailbox, metacache_bundle_download
+
+   Mailbox index bundle file is being downloaded (can happen while the mailbox
+   is being refreshed).
+
+
+.. dovecot_core:event:: metacache_mailbox_bundle_download_finished
+   :added: v2.3.11
+   :inherit: mailbox, metacache_bundle_download
+
+   Mailbox index bundle file was downloaded (can happen while the mailbox is
+   being refreshed).
+
+
+.. dovecot_event:field_group:: metacache_upload
+
+   :field error: Error message if the upload failed.
+
+
+.. dovecot_core:event:: metacache_upload_started
+   :added: v2.3.11
+   :inherit: metacache_upload
+
+   Changes in metacache are being uploaded to storage.
+
+   .. todo:: Does this inherit from mail_user or mailbox?
```

Comment:
Either. Depends on whether metacache root bundle is uploaded or mailbox bundle is being up loaded.